File: shopy-api/Models/ProductImage.cs
namespace shopy_api.Models;

public class ProductImage : ISoftDeletable
{
    public Guid Id { get; set; }

    public Guid ProductId { get; set; }
    public Product Product { get; set; } = null!;

    public string Url { get; set; } = string.Empty;
    public int SortOrder { get; set; }
    public bool IsPrimary { get; set; }

    public bool IsDeleted { get; set; }
}

File: shopy-seller/lib/models/product/product_image.dart
class ProductImage {
  final String id;
  final String url;
  final int sortOrder;
  final bool isPrimary;

  const ProductImage({
    required this.id,
    required this.url,
    required this.sortOrder,
    required this.isPrimary,
  });

  factory ProductImage.fromJson(Map<String, dynamic> json) {
    return ProductImage(
      id: json['id'] as String,
      url: json['url'] as String,
      sortOrder: json['sortOrder'] as int,
      isPrimary: json['isPrimary'] as bool,
    );
  }
}
